migrator/main.go

```diff
@@ -56,38 +61,71 @@ func run() error {
 		return nil
 	}
 
-	// TODO: ROX-9884, ROX-10700 -- turn off replicas and migrations until Postgres updates complete.
-	if !features.PostgresDatastore.Enabled() {
-		dbm, err := replica.Scan(migrations.DBMountPath(), conf.Maintenance.ForceRollbackVersion)
+	if features.PostgresDatastore.Enabled() {
+		sourceMap, adminConfig, err := pgconfig.GetPostgresConfig()
 		if err != nil {
-			return errors.Wrap(err, "fail to scan replicas")
+			return errors.Wrap(err, "unable to get Postgres DB config.")
 		}
+		dbm := postgres.New(conf.Maintenance.ForceRollbackVersion, adminConfig, sourceMap)
 
-		replicaName, replicaPath, err := dbm.GetReplicaToMigrate()
+		// Scan for database replicas
+		err = dbm.Scan()
 		if err != nil {
-			return err
-		}
-		option.MigratorOptions.DBPathBase = replicaPath
-		if err = upgrade(conf); err != nil {
-			return err
-		}
-		if err = dbm.Persist(replicaName); err != nil {
-			return err
+			return errors.Wrap(err, "fail to scan replicas")
 		}
-	} else {
-		var gormDB *gorm.DB
-		gormDB, err := postgreshelper.Load(conf)
-		if err != nil {
-			return errors.Wrap(err, "failed to connect to postgres DB")
+
+		// If we have database replicas we need to process them.  Otherwise, we can assume we are
+		// starting from scratch and as such we will need to create the DB and apply schemas with Gorm.
+		if len(dbm.ReplicaMap) > 0 {
```

Comment:
We may need to check both postgres and rocksdb for replicas.
We have a few scenarios here:
1. This is the first time we migrate from Rocksdb -> Postgresdb, we have the flag enabled but we actually need to get all replicas from rocksdb.
2. We had tried to upgrade to Rocksdb -> Postgresdb before but failed somewhere, so we may have most replicas in rocksdb but some residues in Postgres DB.
3. We have upgraded to Postgres before any changes are persisted successfully.
We can tell such conditions by checking if the action is in Postgres or in Rocksdb or both.